app/services/catalog/create_approval_request.rb

```diff
@@ -47,11 +47,12 @@ def workflows(portfolio_item)
       portfolio_item.resolved_workflow_refs
     end
 
-    def create_approval_request(req)
+    def create_approval_request(order_item, req)
       ApprovalRequest.new.tap do |approval|
         approval.workflow_ref         = req.workflow_id
         approval.approval_request_ref = req.id
         approval.state                = req.decision.to_sym
+        approval.tenant_id            = order_item.tenant_id
```

Comment:
👍

Reply:
@syncrou Since this is in the path of the controller wouldn't the ActAsTenant Gem already have set the tenant for the record creation?
https://github.com/ManageIQ/catalog-api/blob/b2ed5b108af4f514bcf2d2d5f9b8843d3d06af40/app/controllers/application_controller.rb#L13
If we have to do this for every record its going to be painful we might miss some. I was under the impression that the with_request would set the with_tenant inside of it.

Reply:
Yes you're correct about `ActsAsTenant`, I did it this way just to make sure to set the approval's tenant to the same as the order_item - in case an admin (or just anyone who didn't create it) made the order request. Unless we don't have to worry about that

Reply:
@lindgrenj6 Lets remove it since its redundant, since this call is made from the controller it would always have it, I am trying to figure out how to avoid setting this for every object. The only path that is not covered are the ones that don't go thru the controller like the listener which  is not true here

Reply:
Ok, I reverted that change, the tenant will get set in the controller just like the other models.